src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java

```diff
@@ -1255,7 +1254,11 @@ private QueueEntry dispatchOperation(JedisCluster jedis, List<Platform.Property>
                 operationName, operationQueue.getDequeueName()));
       }
       dispatchedOperations.remove(jedis, operationName);
-      return queueEntry;
+
+      // Return an entry so that if it needs re-queued, it will have the correct "requeue attempts".
+      QueueEntry entryForRequeue =
+          queueEntryBuilder.setRequeueAttempts(queueEntryBuilder.getRequeueAttempts() + 1).build();
```

Comment:
Generally considered bad practice to get this value out of the builder. I would also ask, given that this logic was wrong, that you write a test to ensure that for established transitions (i.e. queued -> dispatched -> expiration -> requeue) that you see the value go from 0, to 1, to 2.

Reply:
good idea.  I tested this from a backplane API